backend/game_server/lib/entity.ex

```diff
@@ -0,0 +1,46 @@
+defmodule BernacleServer.Entity do
+	alias BernacleServer.Entity
+
+	defstruct id: 0, attributes: %{}, abilities: %{}
+
+	def start_link(en = %Entity{}), do: Agent.start_link(fn -> en end) 
+
+	def new, do: %Entity{}
+
+	def kill(en), do: Agent.stop(en)
+
+	def get_attribute(%Entity{attributes: attr}, key), do: Map.get(attr, key)
+
+	def get_ability(%Entity{abilities: abi}, key), do: Map.get(abi, key)
```

Comment:
aren't we except that all actions are don't on agents, that all entities are separate processes?
if yes, I believe this direct manipulating function should be private to avoid mistakes. `defp`